src/py/flwr/superexec/simulation.py

```diff
@@ -0,0 +1,139 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Simulation engine executor."""
+
+
+import subprocess
+import sys
+from logging import ERROR, INFO
+from pathlib import Path
+from typing import Dict, Optional
+
+from typing_extensions import override
+
+from flwr.cli.config_utils import get_fab_metadata
+from flwr.cli.install import install_from_fab
+from flwr.common.constant import RUN_ID_NUM_BYTES
+from flwr.common.logger import log
+from flwr.server.superlink.state.utils import generate_rand_int_from_bytes
+
+from .executor import Executor, RunTracker
+
+
+class SimulationEngine(Executor):
+    """Simulation engine executor.
+
+    Parameters
+    ----------
+    num_supernodes: Opitonal[str] (default: None)
+        Total number of nodes to involve in the simulation.
+    """
+
+    def __init__(
+        self,
+        num_supernodes: Optional[str] = None,
+    ) -> None:
+        self.num_supernodes = num_supernodes
+
+    @override
+    def set_config(
+        self,
+        config: Dict[str, str],
+    ) -> None:
+        """Set executor config arguments.
+
+        Parameters
+        ----------
+        config : Dict[str, str]
+            A dictionary for configuration values.
+            Supported configuration key/value pairs:
+            - "num-supernodes": str
+                Number of nodes to register for the simulation.
+        """
+        if not config:
+            return
+        if num_supernodes := config.get("num-supernodes"):
+            self.num_supernodes = num_supernodes
+
+    @override
+    def start_run(
+        self, fab_file: bytes, override_config: Dict[str, str]
+    ) -> Optional[RunTracker]:
+        """Start run using the Flower Simulation Engine."""
+        try:
+            if self.num_supernodes is None:
+                log(
+                    ERROR,
+                    "To start a run with the simulation plugin, please specify "
+                    "the number of supernodes. You can do this by using the "
+                    "`--executor-config` argument when launching the SuperExec.",
+                )
+                raise ValueError("Need to set `num-supernodes`.")
+
+            if override_config:
+                raise ValueError(
+                    "Overriding the run config is not yet supported with the "
+                    "simulation executor.",
+                )
+
+            # Install FAB to flwr dir
+            _, fab_id = get_fab_metadata(fab_file)
+            fab_path = install_from_fab(fab_file, None, True)
+
+            # Install FAB Python package
+            subprocess.check_call(
+                [sys.executable, "-m", "pip", "install", str(fab_path)],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
```

Comment:
Can we unify this with the implementation in `DeploymentExecutor`? For example, in `DeploymentExecutor`, we now added `--no-deps`.